lib/utils/utf8.c

```diff
@@ -79,3 +79,32 @@ char *utf8_lcpy(char *dst, const char *src, size_t n)
 
 	return dst;
 }
+
+size_t utf8_count_chars(const char *s) {
+    size_t count = 0;
+    const unsigned char *p = (const unsigned char *)s;
+
+    while (*p) {
+        if ((*p & 0x80) == 0) {
+            // 1-byte character: 0xxxxxxx
```

Comment:
Use `/* ... */` style comments.

```suggestion
            /* 1-byte character: 0xxxxxxx */
```